### PR TITLE
Add docs for writing Mermaid diagrams

### PR DESCRIPTION
**Pull request by ghoshnirmalya**
## Demo
Check out the changes at https://mintlify-nirmalya-add-support-for-mermaid-diagrams.mintlify.app/mermaid.

## Screenshot

|Preview of the whole page|
|-|
|image|

### CODE REVIEW

File: mermaid.mdx
Comment (handotdev) [inline mermaid.mdx:1]: Can we move it into the components tab and have it with the components page style (request examples)?

File: mermaid.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: "Mermaid diagrams"
+description: "Display diagrams using [Mermaid](https://mermaid.js.org/)"
```
Comment (handotdev) [inline mermaid.mdx]: Don't need two links that reference mermaid.js. Can probably remove the link on the description

File: mermaid.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        top2[top] --> bottom2[bottom]
+    end
+    %% ^ These subgraphs are identical, except for the links to them:
+
+    %% Link *to* subgraph1: subgraph1 direction is maintained
+    outside --> subgraph1
+    %% Link *within* subgraph2:
+    %% subgraph2 inherits the direction of the top-level graph (LR)
+    outside ---> top2
+```
```
Comment (handotdev) [inline mermaid.mdx]: Recommend moving this to the top. Ditto on the components style

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
Can we move this content into the Components tab? I think it's more suitable there.

File: mint.json [hunk 3]
```diff
@@ -146,7 +146,8 @@
         "content/components/frames",
         "content/components/steps",
         "content/components/tabs",
-        "content/components/tooltips"
+        "content/components/tooltips",
+        "content/components/mermaid-diagrams"
```
Comment (handotdev) [inline mint.json]: Can we order it alphabetically?

File: content/components/mermaid-diagrams.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        top2[top] --> bottom2[bottom]
+    end
+    %% ^ These subgraphs are identical, except for the links to them:
+
+    %% Link *to* subgraph1: subgraph1 direction is maintained
+    outside --> subgraph1
+    %% Link *within* subgraph2:
+    %% subgraph2 inherits the direction of the top-level graph (LR)
+    outside ---> top2
+```
```
Comment (handotdev) [inline content/components/mermaid-diagrams.mdx]: Oh I also mean having this (the generated example) at the top

File: mint.json (hunk 3 above)
Comment (ghoshnirmalya) [reply mint.json]: Done.

Screenshot 2024-03-08 at 5 58 37 AM

File: content/components/mermaid-diagrams.mdx (hunk 4 above)
Comment (ghoshnirmalya) [reply content/components/mermaid-diagrams.mdx]: Done.

image

File: content/components/mermaid-diagrams.mdx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+````
```
Comment (handotdev) [inline content/components/mermaid-diagrams.mdx]: Don't need this. It's a duplicate since we already have it in the request example

**Review by hahnbeelee — APPROVED** [review]
lgtm

**Review by handotdev — APPROVED** [review]
Nice!!!

**Pull request merged**